pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -159,15 +165,15 @@ type Image struct {
 type BuildStatus struct {
 	// The Register status of the Build
 	// +optional
-	Registered corev1.ConditionStatus `json:"registered,omitempty"`
+	Registered *corev1.ConditionStatus `json:"registered,omitempty"`
 
 	// The reason of the registered Build, it's an one-word camelcase
 	// +optional
-	Reason BuildReason `json:"reason,omitempty"`
+	Reason *BuildReason `json:"reason,omitempty"`
 
 	// The message of the registered Build, either an error or succeed message
 	// +optional
-	Message string `json:"message,omitempty"`
+	Message *string `json:"message,omitempty"`
```

Comment:
Optional string where we are not required to distinguish `nil` and `""`. See discussion above.